app/uk/gov/hmrc/exports/services/reversemapping/declaration/ExportsDeclarationXmlParser.scala

```diff
@@ -16,49 +16,53 @@
 
 package uk.gov.hmrc.exports.services.reversemapping.declaration
 
-import uk.gov.hmrc.exports.models.DeclarationType
-import uk.gov.hmrc.exports.models.declaration._
-import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser.XmlParserResult
-import uk.gov.hmrc.exports.services.reversemapping.declaration.items.ItemsParser
-
 import java.time.Instant
 import java.util.UUID
-import javax.inject.Inject
+
 import scala.xml.NodeSeq
 
+import javax.inject.Inject
+import uk.gov.hmrc.exports.models.DeclarationType._
+import uk.gov.hmrc.exports.models.declaration.AdditionalDeclarationType._
+import uk.gov.hmrc.exports.models.declaration._
+import uk.gov.hmrc.exports.services.reversemapping.DeclarationId
+import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser.XmlParserResult
+import uk.gov.hmrc.exports.services.reversemapping.declaration.items.ItemsParser
+
 class ExportsDeclarationXmlParser @Inject()(
   additionalDeclarationTypeParser: AdditionalDeclarationTypeParser,
   consignmentReferencesParser: ConsignmentReferencesParser,
+  linkDucrToMucrParser: LinkDucrToMucrParser,
   mucrParser: MucrParser,
   itemsParser: ItemsParser
 ) {
 
-  def fromXml(xml: String): XmlParserResult[ExportsDeclaration] = {
+  def fromXml(declarationId: DeclarationId, xml: String): XmlParserResult[ExportsDeclaration] = {
     val declarationXml = scala.xml.XML.loadString(xml)
-
-    buildExportsDeclaration(declarationXml)
+    buildExportsDeclaration(declarationId, declarationXml)
   }
 
-  private def buildExportsDeclaration(declarationXml: NodeSeq): XmlParserResult[ExportsDeclaration] =
+  private def buildExportsDeclaration(declarationId: DeclarationId, declarationXml: NodeSeq): XmlParserResult[ExportsDeclaration] =
     for {
       additionalDeclarationType <- additionalDeclarationTypeParser.parse(declarationXml)
+      declarationType <- deriveDeclarationType(additionalDeclarationType)
       consignmentReferences <- consignmentReferencesParser.parse(declarationXml)
+      linkDucrToMucr <- linkDucrToMucrParser.parse(declarationXml)
       mucr <- mucrParser.parse(declarationXml)
       items <- itemsParser.parse(declarationXml)
-
     } yield
       ExportsDeclaration(
         id = UUID.randomUUID().toString,
-        eori = "GB1234567890",
+        eori = declarationId.eori,
         status = DeclarationStatus.COMPLETE,
         createdDateTime = Instant.now(),
         updatedDateTime = Instant.now(),
         sourceId = None,
-        `type` = DeclarationType.STANDARD,
+        `type` = declarationType,
         dispatchLocation = None,
         additionalDeclarationType = additionalDeclarationType,
-        consignmentReferences = consignmentReferences,
-        linkDucrToMucr = None,
+        consignmentReferences = consignmentReferences.map(_.copy(mrn = Some(declarationId.mrn))),
```

Comment:
I don't think setting the MRN is actually part of this ticket (only `lrn` and `ducr` consignmentReferences fields are mentioned). Probably better to leave it as None.

Reply:
Problem is that the MRN is not part of any of the `XML mapping` tickets. Still I think it does not hurt to have it as reminder of what we are supposed to do if the data cannot be directly derived from parsing the xml.

Reply:
Actually the MRN we'll be sending to DIS is not the same one in ConsignmentReferences, so I'll leave it to None :+1: